docs/DeveloperGuide.md

```diff
@@ -296,6 +296,151 @@ The following sequence diagram shows how the flow of the example execution:
+
+The second stage is to parse each Parameter and flags. This work is delegated to the `typeParsingUtil` class, which is a utility class that contains many static methods that are responsible for parsing different types of user input that is reused across different command parsers.
+
+Then, the `EditPersonCommandParser` class is responsible for combining the results of the previous stage and construct the `EditPersonCommand` object.
+
+Finally, the `EditPersonCommand` class is responsible for executing the command and updating the model accordingly.
+
+The following section will discuss the implementation of the later three groups of classes in more detail.
+
+##### 1. ListEntry and ListEntryField class
+In TutorMates, Lesson and Person extends `ListEntry` class as they are displayed as items of the ___STUDENTS list___, ___SCHEDULE list___ respectively.
+
+Each `ListEntry` object contains a list of `ListEntryField` objects, which are the fields of the `ListEntry` object. For example, a `Person` object contains a list of `ListEntryField` objects, which are the fields of the `Person` object, such as name, phone number, email address, etc.
+
+These two classes are used in multiple generic class and methods to achieve the flexibility and robustness of the parsing process.
+
+##### 2. Parsing of Flags and Parameters
+The `TypeParsingUtil` class parse all kinds of flags (which are all `ListEntryField` ) via a single powerful generic method, `parseField`. The method signature is as follows:
+```
+ public static <T extends ListEntryField> T parseField(String flagName,
+                                                          String input,
+                                                          Of<T> of,
+                                                          boolean isOptional) throws ParseException
+```
+The first parameter, `flagName`, is the name of the flag that is being parsed. It will also be used to generate the error message when the parsing fails.
+
+The second parameter, `input`, is the user input that is intended to be parsed into the flag.
+
+The third parameter, `of`, is a self-defined functional interface that is to be used as the "factory" to create the flag object. The `of` method signature is as follows:
+```
+public interface Of<T extends ListEntryField> {
+    T apply(String str) throws IllegalArgumentException, ParseException;
+}
+```
+The last parameter, `isOptional`, is a boolean flag that indicates whether the flag is optional or not. If the flag is not found in the user input, this method will throw a `ParseException` if the flag is not optional, but will return a `null` object if the flag is optional. It is overloaded to has a default value of `true` when omitted.
```

Comment:
has -> have

Reply:
Thank you for pointing that out. Fixed in the latest commit.